src/plots/gl2d/scene2d.js

```diff
@@ -286,10 +292,11 @@ proto.updateFx = function(options) {
 
     fullLayout.dragmode = options.dragmode;
     fullLayout.hovermode = options.hovermode;
-};
 
-var relayoutCallback = function(scene) {
+    this.graphDiv._fullLayout = fullLayout;
```

Comment:
this seems a bit scary - why is this necessary? I'm just worried that if `gd._fullLayout` and `scene.fullLayout` have diverged, there could somehow be changes in `gd._fullLayout` that this will throw away.